threadit-api/Services/ThreadService.cs

```diff
@@ -37,6 +38,78 @@ public ThreadService(PostgresDbContext context)
             }
         }
 
+        public async Task<List<Models.ThreadFull>> GetThreadsBySpoolAsync(string spoolName)
+        {
+            Models.Spool? spool = await new SpoolService(new PostgresDbContext()).GetSpoolByNameAsync(spoolName);
+
+            if (spool == null)
+            {
+                throw new Exception("Spool does not exist.");
+            }
+            Models.Thread[] threads = await this.threadRepository.GetThreadsBySpoolAsync(spool.Id);
+            List<Models.ThreadFull> fullThreads = new List<Models.ThreadFull>();
+            for (int i = 0; i < threads.Length; i++)
+            {
```

Comment:
Should this should use UserRepository to access the db?

Reply:
nah. repository should never be used by anything except the service correlated with it. services are the ones who throw errors, and theyre the ones who make sure the data is ready to be processed. so instead of using the repo and then implementing the same checks, just use the service that is there to do exactly that

Reply:
Oh yeah I misread what was happening...